envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotUpdater.kt

```diff
@@ -31,40 +32,75 @@ class SnapshotUpdater(
         properties = properties
     )
 
-    fun start(changes: Flux<List<LocalityAwareServicesState>>): Flux<List<LocalityAwareServicesState>> {
-        // see GroupChangeWatcher
-        return Flux.combineLatest(
```

Comment:
A new value on "onGroupAdded" triggers an update of snapshots in line 45. As stated in description we only need to rebuild global snapshots when there is a change in services.